### DISCUSSION TITLE
Fastest way to render/view? Fast debug/iteration loops

### DISCUSSION THREAD

**Discussion by chendy** [mozman/ezdxf#753] [Q&A]
Hello,

Thanks for this amazing project.

I want to use ezdxf for code-CAD. Scripting drawings.
This relies on frequently rendering/viewing to check progress.

Any suggestions how to render/view as fast as possible?
Perhaps changing some settings, to reduce render quality?
Perhaps using the fastest dxf viewer?
Perhaps exdxf's viewer is the fastest?

: )

**Comment by mozman** [comment]
Ezdxf is a Python project, therefore it is not the fastest way to render DXF files and rendering DXF files is not even a primary design goal of ezdxf.

**Comment by mbway** [comment]
Perhaps you would be better off using an intermediate representation when constructing your CAD plans then use ezdxf to convert to DXF only when exporting. Then rendering would be much simpler (and faster) and you would have many more options.

For 2d this would be relatively simple. For example if you want to use python for scripting could draw using QGraphicsScene in PyQt or any other drawing API of your choice then implement a way to convert those drawing commands to DXF entities using ezdzf when you are finished and want to create a DXF file.

For 3d I would suggest you use a project that already exists such as [CadQuery](https://github.com/CadQuery/cadquery)  as building something like this is very challenging.
